src/Microsoft.Health.Checkpoints/Storage/BaseStorageCheckpointClient.cs

```diff
@@ -0,0 +1,140 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Health.Checkpoints.Storage
+{
+    public class BaseStorageCheckpointClient : ICheckpointClient
```

Comment:
Typically we only use Base if this is going to be abstract and not used as is.  Would suggest renaming StorageCheckpointClient, it will be the default implementation.  If others build on top of they will be named differently.

Reply:
updated name